internal/internal_event_handlers.go

```diff
@@ -350,6 +354,22 @@ func (wc *workflowEnvironmentImpl) Send(msg *protocolpb.Message, opts ...msgSend
 	wc.outbox = append(wc.outbox, outboxEntry{msg: msg, eventPredicate: sendCfg.pred})
 }
 
+func (wc *workflowEnvironmentImpl) getNewSdkName() string {
```

Comment:
```suggestion
func (wc *workflowEnvironmentImpl) getNewSdkNameAndReset() string {
```

A bit pedantic, but a bit rough to see mutations in getters even internal. Don't need to change though.

Reply:
Yeah that's fair. That is why I called it `getNewSdkName` rather than `getSdkName`. Adding `reset`  is fine to